src/taipy/core/task/_task_manager.py

```diff
@@ -15,25 +15,28 @@
 
 from .._manager._manager import _Manager
 from .._orchestrator._abstract_orchestrator import _AbstractOrchestrator
-from .._orchestrator._orchestrator_factory import _OrchestratorFactory
-from .._version._version_manager_factory import _VersionManagerFactory
+from .._repository._v2._abstract_repository import _AbstractRepository
+
+# from .._repository._v2._filesystem_repository import _FileSystemRepository
+from .._version._version_mixin import _VersionMixin
 from ..common._entity_ids import _EntityIds
 from ..common.alias import CycleId, PipelineId, ScenarioId, TaskId
 from ..common.warn_if_inputs_not_ready import _warn_if_inputs_not_ready
 from ..config.task_config import TaskConfig
 from ..data._data_manager_factory import _DataManagerFactory
 from ..exceptions.exceptions import NonExistingTask
-from ..job._job_manager_factory import _JobManagerFactory
-from ..task._task_repository_factory import _TaskRepositoryFactory
 from ..task.task import Task
 
 
-class _TaskManager(_Manager[Task]):
-    _repository = _TaskRepositoryFactory._build_repository()  # type: ignore
+class _TaskManager(_Manager[Task], _VersionMixin):
+
     _ENTITY_NAME = Task.__name__
+    _repository: _AbstractRepository
 
     @classmethod
     def _orchestrator(cls) -> Type[_AbstractOrchestrator]:
+        from .._orchestrator._orchestrator_factory import _OrchestratorFactory
```

Comment:
@joaoandre-avaiga 
To fix the circular import, you only need to move the import at the beginning of the file to this function :D